MigrationTutorial/Models/V2/Consumable.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using Realms;
+
+namespace MigrationTutorial.Models.V2
+{
+    public class Consumable : RealmObject
+    {
+        [PrimaryKey]
```

Comment:
> What's the advantage of using a different `PrimaryKey` here?
None. This was done simply because asking around to some senior engineers in the organization it came up that changing primary key is a common problem where users get confused. So I just did a simple example?
> Can we safely change that or: how do we make sure the new PrimaryKey that has existed before can now safely be used as such? 
In my specific case I know that `ObjectId` can't collide with the type of ProductId that I have. But it's a fair point to show a deeper check against duplicates, however some is already done as answered below.
> DoMigrate for v2 does not do anything with ProductId, can we end up having duplicates there if we already had them before and do we need to solve this during migration?
`V2Utils.DoMigrate` eliminates all object that have colliding ProductId (line 119-134). It simply assumes that if 2 products with the same `ProductId` exist, it was a mistake and the duplicates get removed. So the problem should be already solved.

Reply:
Makes sense. 👍 
About the colliding objects: I guess it depends on what the specific scenario for the user anyway. They have to decide. So I'd just leave it as is and it should be fine, as you said.